src/lib/hooks/useABTesting.js

```diff
@@ -1,18 +1,48 @@
-import { useEffect } from 'react'
-
+import { useEffect, useState } from 'react'
+import { once } from 'lodash'
 import Config from '../../config/config'
 import { fireEvent } from '../analytics/analytics'
+import AsyncStorage from '../utils/asyncStorage'
+import { AB_TESTING } from '../constants/localStorage'
+import logger from '../logger/pino-logger'
+
+const log = logger.child({ from: 'useABTesting' })
+
+const loadPersistedVariants = once(async () => {
+  return (await AsyncStorage.getItem(AB_TESTING)) || {}
+})
 
-const createABTesting = (percentage = Config.abTestPercentage) => {
-  const isCaseA = Math.random() < percentage
-  const ab = isCaseA ? 'A' : 'B'
+const createABTesting = (testName, percentage = Config.abTestPercentage, persistVariant = true) => {
+  const getTestVariant = once(async () => {
+    const tests = await loadPersistedVariants()
+    let test = tests[testName]
+    if (test == null) {
+      test = { name: testName }
+      test.random = Math.random()
```

Comment:
why do you need all that stuff stored ? only the isCaseA will be enough

Reply:
localstorage has limits and we using it a lot for the feed. so we need to optimize any data we're storing inside

Reply:
that's very small amounts of data.
you are right about the feed. the solution will be find better solution for feed. either stop caching or use indexdb